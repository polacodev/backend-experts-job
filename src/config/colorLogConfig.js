import colors from 'colors';

const colorLog = colors.setTheme({
  success: 'green',
  warn: 'yellow',
  info: 'cyan',
  error: 'red',
});

export default colorLog;
