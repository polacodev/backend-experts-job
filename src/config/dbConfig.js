import mongoose from 'mongoose';

const MONGO_URL = 'mongodb://localhost:27017/ExpertsJob';
const database = async () => {
  try {
    await mongoose.connect(MONGO_URL,
      {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        useFindAndModify: false,
      });
    console.log('DB Connection SUCCESSFULLY');
  } catch (error) {
    console.log('DB Connection FAILED');
  }
};

export default database;
