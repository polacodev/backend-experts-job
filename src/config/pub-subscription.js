import { connect } from 'mqtt';
import { MQTTPubSub } from 'graphql-mqtt-subscriptions';

const client = connect('mqtt://test.mosquitto.org', {
  reconnectPeriod: 1000,
});

const pubsub = new MQTTPubSub({
  client,
});
export default pubsub;
// import { PubSub } from 'apollo-server';

// const pubsub = new PubSub();

// export default pubsub;
