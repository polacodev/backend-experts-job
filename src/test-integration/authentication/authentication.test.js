import querystring from 'querystring';

import { API_URL, api } from '../services.config';

describe('authentication component', () => {
  it('get the user authentication', async () => {
    const query = `
    query auth {
      authentication(auth: {email: "userTest@gmail.com", password: "userTest"}) {
        isAuthenticated
      }
    }
    `;

    const response = await api.post(API_URL, querystring.stringify({ query }));
    expect(response.data).toMatchSnapshot();
  });
});
