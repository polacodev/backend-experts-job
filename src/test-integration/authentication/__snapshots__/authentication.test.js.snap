// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentication component get the user authentication 1`] = `
Object {
  "data": Object {
    "authentication": Object {
      "isAuthenticated": false,
    },
  },
  "extensions": Object {
    "operation": null,
  },
}
`;
